CloudyLauncher.java

```diff
@@ -0,0 +1,497 @@
+
+            writer.writeBytes(queryData);
+            writer.flush();
+            writer.close();
+
+            if (connection.getResponseCode() == HttpURLConnection.HTTP_BAD_REQUEST) {
+
+                String errorMessage = "";
+
+                BufferedReader errorReader = new BufferedReader(new InputStreamReader(connection.getErrorStream()));
+                JSONObject errorResponse = new JSONObject(errorReader.readLine());
+
+                if (errorResponse.has("username")) {
+                    errorMessage = errorMessage + "\nUsername: " + errorResponse.getJSONArray("username").getString(0);
+                }
+
+                if (errorResponse.has("password")) {
+                    errorMessage = errorMessage + "\nPassword: " + errorResponse.getJSONArray("password").getString(0);
+                }
+
+                if (errorResponse.has("non_field_errors")) {
+                    errorMessage = errorMessage + "\n" + errorResponse.getJSONArray("non_field_errors").getString(0);
+                }
+
+                setFeedback(errorMessage);
+
+            } else {
+
+                BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()));        
+                String response = reader.readLine();
+                reader.close();
+
+                JSONObject tokenObj = new JSONObject(response);
+                String responseToken = tokenObj.getString("token");                
+                setToken(responseToken);
+
+                setFeedback("User recognised.");
+
+                initialiseGamePanel();
+            }   
+
+        } catch (IOException e) {
+            setFeedback("Check server.");
+
+        }
+    }
+
+    private void attemptUserRegistration(String username, String password, String email, String firstName, String lastName) {
+
+        try {
+            URL url = new URL(baseurl + "/users/");        
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+
+            connection.setRequestMethod("POST");
+            connection.setRequestProperty("Accept", "application/json");
+            String queryData = String.format("username=%s&password=%s&email=%s&first_name=%s&last_name=%s", username, password, email, firstName, lastName);
+
+            connection.setDoOutput(true);
+            DataOutputStream writer = new DataOutputStream(connection.getOutputStream());
+
+            writer.writeBytes(queryData);
+            writer.flush();
+            writer.close();
+
+            if (connection.getResponseCode() == HttpURLConnection.HTTP_BAD_REQUEST) {   
+                String errorMessage = "";
+
+                BufferedReader errorReader = new BufferedReader(new InputStreamReader(connection.getErrorStream()));
+                JSONObject errorResponse = new JSONObject(errorReader.readLine());
+
+                if (errorResponse.has("username")) {
```

Comment:
Is it possible to make a new method to set the error response? As some part of this code is similar to the error response assignment in the previous method..